baseplate/lib/message_queue.py

```diff
@@ -103,22 +134,39 @@ def put(self, message: bytes, timeout: Optional[float] = None) -> None:
         raise TimedOutError
 
     def unlink(self) -> None:
-        """Remove the queue from the system.
-
-        The queue will not leave until the last active user closes it.
-
-        """
         self.queue.unlink()
 
     def close(self) -> None:
-        """Close the queue, freeing related resources.
+        self.queue.close()
 
-        This must be called explicitly if queues are created/destroyed on the
-        fly. It is not automatically called when the object is reclaimed by
-        Python.
 
-        """
-        self.queue.close()
+class InMemoryMessageQueue(MessageQueue):
+    """An in-memory inter process message queue."""
+
+    def __init__(self, name: str, max_messages: int):
+        self.queue: q.Queue = q.Queue(max_messages)
+        self.max_messages = max_messages
+        self.name = name
```

Comment:
The `InMemoryMessageQueue` does not need / use the `name`, so you can remove it.

Reply:
With recent updates it does use it now (a server may have multiple queues, we reference the queues by name), so leaving and resolving this comment